tests/device_tests/cardano/test_sign_tx.py

```diff
@@ -0,0 +1,142 @@
+
+    expected_responses = [
+        messages.ButtonRequest(code=messages.ButtonRequestType.Other)
+        for i in range(len(parameters["input_flow"]))
+    ]
+    expected_responses.append(messages.CardanoSignedTx())
+
+    def input_flow():
+        for sequence in parameters["input_flow"]:
+            yield
+            for action in sequence:
+                if action == "SWIPE":
+                    client.debug.swipe_up()
+                elif action == "YES":
+                    client.debug.press_yes()
+                else:
+                    raise ValueError("Invalid input action")
+
+    with client:
+        client.set_expected_responses(expected_responses)
+        client.set_input_flow(input_flow)
+        response = cardano.sign_tx(
+            client=client,
+            inputs=inputs,
+            outputs=outputs,
+            fee=parameters["fee"],
+            ttl=parameters["ttl"],
+            certificates=certificates,
+            withdrawals=withdrawals,
+            metadata=bytes.fromhex(parameters["metadata"]),
+            protocol_magic=parameters["protocol_magic"],
+            network_id=parameters["network_id"],
+        )
+        assert response.tx_hash.hex() == result["tx_hash"]
+        assert response.serialized_tx.hex() == result["serialized_tx"]
+
+
+@parametrize_using_common_fixtures("cardano/sign_tx.failed.json")
+def test_cardano_sign_tx_failures(client, parameters, result):
+    inputs = [cardano.create_input(i) for i in parameters["inputs"]]
+    outputs = [cardano.create_output(o) for o in parameters["outputs"]]
+    certificates = [cardano.create_certificate(c) for c in parameters["certificates"]]
+    withdrawals = [cardano.create_withdrawal(w) for w in parameters["withdrawals"]]
+
+    expected_responses = [messages.Failure()]
+
+    with client:
+        client.set_expected_responses(expected_responses)
+
+        with pytest.raises(TrezorFailure, match=result["error_message"]):
+            cardano.sign_tx(
+                client=client,
+                inputs=inputs,
+                outputs=outputs,
+                fee=parameters["fee"],
+                ttl=parameters["ttl"],
+                certificates=certificates,
+                withdrawals=withdrawals,
+                metadata=bytes.fromhex(parameters["metadata"]),
+                protocol_magic=parameters["protocol_magic"],
+                network_id=parameters["network_id"],
+            )
+
+
+@parametrize_using_common_fixtures("cardano/sign_tx.slip39.json")
+def test_cardano_sign_tx_slip39(client, parameters, result):
+    inputs = [cardano.create_input(i) for i in parameters["inputs"]]
+    outputs = [cardano.create_output(o) for o in parameters["outputs"]]
+    certificates = [cardano.create_certificate(c) for c in parameters["certificates"]]
+    withdrawals = [cardano.create_withdrawal(w) for w in parameters["withdrawals"]]
+
+    expected_responses = [messages.PassphraseRequest()]
```

Comment:
is this identical to `test_cardano_sign_tx` except for the PassphraseRequest? if yes, let's merge that and use `passphrase: ""`

Reply:
Fixed in a1c2b35699dfa759cc8ea045b0254f8984186133.